webapp/src/components/AssessmentsPage.tsx

```diff
@@ -0,0 +1,410 @@
+import {
+  Box,
+  Chip,
+  Container,
+  Stack,
+  Button,
+  Paper,
+  TableRow,
+  TableHead,
+  TableContainer,
+  TableCell,
+  TableBody,
+  Table,
+  Tabs,
+  Tab,
+} from '@mui/material';
+
+import React, { useState } from 'react';
+import Badge, { BadgeProps } from '@mui/material/Badge';
+import { styled } from '@mui/material/styles';
+
+import CheckOutlinedIcon from '@mui/icons-material/CheckOutlined';
+import DoneAllOutlinedIcon from '@mui/icons-material/DoneAllOutlined';
+import ScheduleOutlinedIcon from '@mui/icons-material/ScheduleOutlined';
+import CancelOutlinedIcon from '@mui/icons-material/CancelOutlined';
+import ArchiveOutlinedIcon from '@mui/icons-material/ArchiveOutlined';
+import UpcomingOutlinedIcon from '@mui/icons-material/UpcomingOutlined';
+import LockClockOutlinedIcon from '@mui/icons-material/LockClockOutlined';
+import AssessmentIcon from '@mui/icons-material/Assessment';
+import { formatDateTime } from '../helpers/dateTime';
+import { assessmentList } from '../assets/data';
+
+enum StatusTab {
+  All,
+  Active,
+  Past,
+  Upcoming,
+}
+
+interface TableCellWrapperProps {
+  children?: React.ReactNode;
+  index: number[];
+  statusTab: number;
+}
+interface TableRowWrapperProps {
+  children?: React.ReactNode;
+  statusTab: number;
+  status: string;
+}
+
+function TableCellWrapper(props: TableCellWrapperProps) {
+  const { children, statusTab, index } = props;
+  return index.includes(statusTab) ? <TableCell>{children}</TableCell> : null;
+}
+
+function TableRowWrapper(props: TableRowWrapperProps) {
+  const { children, status, statusTab } = props;
+  if (statusTab === StatusTab.All) {
+    return <TableRow>{children}</TableRow>;
+  }
+  switch (status) {
+    case 'Active':
+      return statusTab === StatusTab.Active ? (
+        <TableRow>{children}</TableRow>
+      ) : null;
+    case 'Upcoming':
+      return statusTab === StatusTab.Upcoming ? (
+        <TableRow>{children}</TableRow>
+      ) : null;
+    case 'Submitted':
+    case 'Graded':
+    case 'Unsubmitted':
+    default:
+      return statusTab === StatusTab.Past ? (
+        <TableRow>{children}</TableRow>
+      ) : null;
+  }
+}
+
+function renderButtonByStatus(status: string) {
+  switch (status) {
+    case 'Active':
+      return (
+        <Button variant="contained" size="small">
+          Start
+        </Button>
+      );
+    case 'Submitted':
+    case 'Graded':
+      return (
+        <Button variant="contained" size="small">
+          View
+        </Button>
+      );
+    case 'Upcoming':
+    case 'Unsubmitted':
```

Comment:
Please see my earlier review comment about being able to see questions and answers for unsubmitted assessments.